### PR TITLE
LPS-184221 Make submit button for workflow readonly if status is pending

### PR DESCRIPTION
**Pull request by igor-franca**
Related [Ticket](https://issues.liferay.com/browse/LPS-184221)

Please, make sure that all business rules are respected with this modification. Thank you in advance

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by igor-franca** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java [hunk 1]
```diff
@@ -329,8 +329,8 @@ public boolean isPublishButtonDisabled() {
 			_dlRequestHelper.getDLPortletInstanceSettings();
 
 		if (_fileEntryDisplayContextHelper.isCheckedOutByOther() ||
-			(_fileVersionDisplayContextHelper.isPending() &&
-			 dlPortletInstanceSettings.isEnableFileEntryDrafts())) {
+			_fileVersionDisplayContextHelper.isPending() ||
+			dlPortletInstanceSettings.isEnableFileEntryDrafts()) {
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java:333]: This looks wrong. You're disabling the button when:
1. It was checked out by a different user,
2. It is pending,
3. or the portlet was configured to support drafts.

If you configure the portlet to support drafts, then the button will be disabled for all documents, no matter their status.

Please, check that your solution works with drafts both enabled and disabled.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java (hunk 1 above)
Comment (igor-franca) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java:333]: Thanks a lot for the quick review.

This makes sense, but I would like to understand better about the relationship between support drafts and the disabled button.

Regardless of whether the support drafts property is enabled or disabled, this button will only be disabled when the status is peding. Correct? If so, I don't see the need to validate through isEnableFileEntryDrafts. I thought of this possible solution:

`
if (_fileEntryDisplayContextHelper.isCheckedOutByOther() ||
			_fileVersionDisplayContextHelper.isPending()) {
			return true;
		}
`

Does it make sense to you?

**Comment by igor-franca** [comment]
ci:test:sf

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java]: Yes, that looks better. That code is pretty old; that condition may have been introduced before wf was even a thing for DM.

**Review by adolfopa — APPROVED** [review]

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:forward

**Pull request closed without merging**